src/main/java/com/vackosar/gitflowincrementalbuild/boundary/UnchangedProjectsRemover.java

```diff
@@ -309,4 +364,38 @@ private static boolean matchesSelector(MavenProject project, String selector, Fi
             return false;
         }
     }
+
+    private static class LazyExpressionEvaluator {
+
+        private static final Logger LOGGER = LoggerFactory.getLogger(LazyExpressionEvaluator.class);
+
+        private final MavenSession session;
+        private final MavenProject project;
+
+        private TypeAwareExpressionEvaluator evaluator;
+
+        public LazyExpressionEvaluator(MavenSession session, MavenProject project) {
+            this.session = session;
+            this.project = project;
+        }
+
+        public String evaluate(String expression) {
+            if (expression == null || !expression.contains("${")) {
+                return expression;
+            }
+            if (evaluator == null) {
+                // set project on cloned session otherwise properties might be resolved from a more or less unrelated project
+                MavenSession clonedSession = session.clone();
+                clonedSession.setCurrentProject(project);
+                // there is also a ctor without MojoExecution parameter but evaluate() will then fail with a NPE (sic!)
```

Comment:
Did you file an issue for that? Reference would be great for later removal of this workaround.

Reply:
Not yet. I'll check briefly.

Reply:
https://issues.apache.org/jira/browse/MNG-6982